frontend/dockerfile/builder/build.go

```diff
@@ -362,7 +362,7 @@ func Build(ctx context.Context, c client.Client) (*client.Result, error) {
 					}
 				}
 				// old API
-				if cacheFromStr := opts[keyCacheFrom]; cacheFromStr != "" {
+				if cacheFromStr := opts[keyCacheFrom]; len(cacheImports) < 1 && cacheFromStr != "" {
```

Comment:
what is the purpose of this change?

Reply:
Yes, this is needed, since both "cache-imports" and "cache-from" will be passed from client to compatible with legacy frontend (prefer to cache-imports but cache-from is given as well), we need to avoid the duplicated `CacheOptionsEntry`.